src/Grid/grid.jl

```diff
@@ -68,12 +68,133 @@ struct VertexIndex <: BoundaryIndex
     idx::Tuple{Int,Int} # cell and side
 end
 
+struct Neighbor{T<:BoundaryIndex}
+    neighbor_info::Vector{T}
+end
+
+Neighbor(info::T) where T <: BoundaryIndex = Neighbor([info])
+Base.zero(::Type{Neighbor}) = Neighbor(BoundaryIndex[])
+Base.length(n::Neighbor) = length(n.neighbor_info)
+Base.getindex(n::Neighbor,i) = getindex(n.neighbor_info,i)
+Base.firstindex(n::Neighbor) = 1
+Base.lastindex(n::Neighbor) = length(n.neighbor_info)
+Base.:(==)(n1::Neighbor, n2::Neighbor) = n1.neighbor_info == n2.neighbor_info
+
+function Base.:+(n1::Neighbor, n2::Neighbor)
+    neighbor_info = [n1.neighbor_info; n2.neighbor_info]
+    return Neighbor(neighbor_info)
+end
+
+function Base.show(io::IO, ::MIME"text/plain", n::Neighbor)
+    if length(n) == 0
+        println(io, "No Neighbor")
+    elseif length(n) == 1
+        println(io, "$(n.neighbor_info[1])")
+    else
+        println(io, "$(n.neighbor_info...)")
+    end
+end
+
+face_npoints(::Cell{2,N,M}) where {N,M} = 2
+face_npoints(::Cell{3,4,1}) = 4 #not sure how to handle embedded cells e.g. Quadrilateral3D
+edge_npoints(::Cell{3,4,1}) = 2 #not sure how to handle embedded cells e.g. Quadrilateral3D
+face_npoints(::Cell{3,N,6}) where N = 4
+face_npoints(::Cell{3,N,4}) where N = 3
+edge_npoints(::Cell{3,N,M}) where {N,M} = 2
+
+getdim(::Cell{dim,N,M}) where {dim,N,M} = dim
+
+abstract type AbstractTopology end
+
+struct GridTopology <: AbstractTopology
+    face_neighbor::SparseMatrixCSC{Neighbor,Int}
+    corner_neighbor::SparseMatrixCSC{Neighbor,Int}
+    edge_neighbor::SparseMatrixCSC{Neighbor,Int}
```

Comment:
Names are a bit weird. I have no good alternative for now. 

Depending on the exact definitions of these tables, maybe some of the "inverse relations" are missing (e.g. vertex -> cell)

Reply:
hm I don't know how to call them, `*_neighbor` is a matrix that maps a cell and a local id to a `*Index` where `*Index` is the exclusive neighborhood connection to the given cell and local id

Reply:
Maybe [x]_to_entity_table?